src/python/pants/engine/internals/graph.py

```diff
@@ -678,19 +701,33 @@ async def resolve_dependencies(
                 inference_request_type(sources_field),
             )
 
-    original_addresses = {*provided.addresses, *itertools.chain.from_iterable(injected)}
-    generated_addresses = {*inferred, *itertools.chain.from_iterable(explicit_file_deps_owners)}
+    original_addresses = {
+        addr
+        for addr in (*provided.addresses, *itertools.chain.from_iterable(injected))
+        if addr not in provided.ignored_addresses
+    }
+    flattened_ignore_file_deps_owners = set(
+        itertools.chain.from_iterable(explicit_file_deps_ignore_owners)
+    )
+    generated_addresses = {
+        addr
+        for addr in (*inferred, *itertools.chain.from_iterable(explicit_file_deps_owners))
+        if addr not in flattened_ignore_file_deps_owners
+    }
     # If a generated subtarget's original base target is already included, then we leave off the
     # generated subtarget. We also leave it off if the generated subtarget's original base target
-    # is the target we're resolving dependencies for.
-    remaining_generated_addresses = {
-        addr
-        for addr in generated_addresses
+    # is the target we're resolving dependencies for. Finally, we leave it off if the user said to
+    # ignore the original owning target.
```

Comment:
It might be good to validate from day one that a `!` is always used (by confirming that that dep was actually going to be in the set without the exclude)... since this is a brand new feature, we could be more opinionated about requiring it (or having it be only warn/error, with no silent option).